pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotTableRestletResource.java

```diff
@@ -950,6 +950,40 @@ public String getTableAggregateMetadata(
     return segmentsMetadata;
   }
 
+  @GET
+  @Path("tables/{tableName}/validDocIdMetadata")
+  @Authorize(targetType = TargetType.TABLE, paramName = "tableName", action = Actions.Table.GET_METADATA)
+  @Produces(MediaType.APPLICATION_JSON)
+  @ApiOperation(value = "Get the aggregate valid doc id metadata of all segments for a table", notes = "Get the "
+      + "aggregate valid doc id metadata of all segments for a table")
+  public String getTableAggregateValidDocIdMetadata(
+      @ApiParam(value = "Name of the table", required = true) @PathParam("tableName") String tableName,
+      @ApiParam(value = "OFFLINE|REALTIME") @QueryParam("type") String tableTypeStr) {
```

Comment:
The table type should always be `REALTIME` as this API makes sense only for upsert tables. Should we just remove this from request?

Reply:
@tibrewalpratik17 Currently, I return the error when the table is `OFFLINE`. In the future, we may want to support upsert in the offline table. We can use this flag once we support this. I'm trying to follow the convention in this class to make the API look similar as other ones.